### PR TITLE
move to nom7

### PR DESCRIPTION
**Pull request by zuisong**
- move to nom7 🥳
- upgrade deps

### CODE REVIEW

**Comment by adwhit** [comment]
Hi - thanks for this.

I just pulled your changes and tried to run the `connect.rs` example against activemq docker and unfortunately I got parser failures. Does the same happen for you?

**Comment by zuisong** [comment]
The tests I conducted under RabbitMQ ran smoothly. Let me try running them under ActiveMQ and see how it performs.

**Comment by zuisong** [comment]
I have tested it under the following container, and it is running normally.
`docker run --detach --name activemq-artemis  --network=host  --rm apache/activemq-artemis:latest-alpine`

default username and password is `artemis` 
It is possible that the account credentials you used for the connection were incorrect.

**Comment by zuisong** [comment]
I have reproduced this issue with `apache/activemq-classic` image.

server return CONNECTED frame with
```
CONNECTED
server:ActiveMQ/6.0.0
heart-beat:0,0
session:ID:orbstack-45879-1702220142549-3:2
version:1.2

```
In the STOMP protocol specification, colons are not allowed.
```
header-name         = 1*<any OCTET except CR or LF or ":">
header-value        = *<any OCTET except CR or LF or ":">
```


Perhaps we should consider accommodating this situation?

**Comment by zuisong** [comment]
@adwhit  I have added code to handle this situation. Please try again.

File: src/frame.rs [hunk 1]
```diff
@@ -123,33 +116,42 @@ fn is_empty_slice(s: &[u8]) -> Option<&[u8]> {
...
-            >> tag!("\x00")
-            >> many0!(complete!(eol))
-            >> (Frame {
-                command,
-                headers,
-                body,
-            })
-    )
-);
+pub fn parse_frame(input: &[u8]) -> IResult<&[u8], Frame> {
```
Comment (adwhit) [inline src/frame.rs]: this should be `pub(crate)`. Currently this fails to compile on `rustc 1.72.1` although interestingly it works for `rustc 1.74.0`

**Review by adwhit — CHANGES_REQUESTED** [review]
Interesting that the 'classic' docker image is in violation of the spec - nevertheless I think it is good to be permissive so thank you for the fix.

Just one comment then we should be good to merge

File: src/client.rs [hunk 2]
```diff
@@ -17,15 +17,16 @@ use anyhow::{anyhow, bail};
 /// If successful, returns a tuple of a message stream and a sender,
 /// which may be used to receive and send messages respectively.
 pub async fn connect(
-    address: impl Into<String>,
+    server: impl Into<String>,
```
Comment (adwhit) [inline src/client.rs]: this should be `impl ToSocketAddrs`, which can then be passed directly to `ClientCode.framed`

File: src/client.rs [hunk 3]
```diff
@@ -17,15 +17,16 @@ use anyhow::{anyhow, bail};
...
+    host: impl Into<String>,
     login: Option<String>,
     passcode: Option<String>,
 ) -> Result<ClientTransport> {
-    let address = address.into();
+    let address: String = server.into();
     let addr = address.as_str().to_socket_addrs().unwrap().next().unwrap();
```
Comment (adwhit) [inline src/client.rs:26]: rather than `unwrap()`, this should return an error if it the conversion fails

**Comment by zuisong** [comment]
It should be fixed now. 😄

**Comment by adwhit** [comment]
Thanks!

**Comment by adwhit** [comment]
This PR has been released as `v0.4.0`

**Pull request merged**